src/Mod/Path/PathScripts/PathProfileEdges.py

```diff
@@ -1,105 +1,580 @@
-                for sub in b[1]:
-                    edgelist.append(getattr(b[0].Shape, sub))
-                basewires.append((b[0], findWires(edgelist)))
-                if zMin is None or b[0].Shape.BoundBox.ZMin < zMin:
-                    zMin = b[0].Shape.BoundBox.ZMin
-
-            for base,wires in basewires:
-                for wire in wires:
-                    f = Part.makeFace(wire, 'Part::FaceMakerSimple')
-
-                    # shift the compound to the bottom of the base object for
-                    # proper sectioning
-                    zShift = zMin - f.BoundBox.ZMin
-                    newPlace = FreeCAD.Placement(FreeCAD.Vector(0, 0, zShift), f.Placement.Rotation)
-                    f.Placement = newPlace
-                    env = PathUtils.getEnvelope(base.Shape, subshape=f, depthparams=self.depthparams)
-                    shapes.append((env, False))
-        return shapes
-
-def SetupProperties():
-    return PathProfileBase.SetupProperties()
-
-def Create(name, obj = None):
-    '''Create(name) ... Creates and returns a Profile based on edges operation.'''
-    if obj is None:
-        obj = FreeCAD.ActiveDocument.addObject("Path::FeaturePython", name)
-    obj.Proxy = ObjectProfile(obj, name)
-    return obj
+# -*- coding: utf-8 -*-
+
+# ***************************************************************************
+# *                                                                         *
+# *   Copyright (c) 2016 sliptonic <shopinthewoods@gmail.com>               *
+# *                                                                         *
+# *   This program is free software; you can redistribute it and/or modify  *
+# *   it under the terms of the GNU Lesser General Public License (LGPL)    *
+# *   as published by the Free Software Foundation; either version 2 of     *
+# *   the License, or (at your option) any later version.                   *
+# *   for detail see the LICENCE text file.                                 *
+# *                                                                         *
+# *   This program is distributed in the hope that it will be useful,       *
+# *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+# *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+# *   GNU Library General Public License for more details.                  *
+# *                                                                         *
+# *   You should have received a copy of the GNU Library General Public     *
+# *   License along with this program; if not, write to the Free Software   *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+# *   USA                                                                   *
+# *                                                                         *
+# ***************************************************************************
+
+import FreeCAD

```

Comment:
This code seems to be identical to the previously existing code, so what is actually changing? It's possible that you are modifying the line endings? These should be Unix line endings and not Windows line endings. Double check your editor.

Reply:
Thanks for the 'line endings' suggestion.  I've found where I introduced the related error.  It has been fixed.  Having mentioned this, we might want the GitHub devs with commit privileges to go through their respective workbenches and change all files with non-Unix endings to the Unix style.  This would actually shrink the file sizes just a little where Windows style are currently used.
 
Of the two files affected in this commit, one is Windows style and the other Unix.

Thanks.

Reply:
This was discussed in the forum, [PR 2730, "Fix" line-endings](https://forum.freecadweb.org/viewtopic.php?f=17&t=41117), #2730. Also see #2751, #2757, #2789.

It was already implemented for several workbenches, like FEM, AddonManager, Draft, Arch, Material.

Basically, you can provide a `.gitattributes` file in the root of `src/Mod/Path`, so that all further commits that you do automatically change the line endings to the desired Unix endings. See a file like that in 7f3f71f17.

Reply:
On the topic of the line endings, the mentioned #2789 added the `.gitattributes` file to some directories, however, the subsequent #2796 removed them again. Instead, a global `.gitattributes` file is provided for the entire `src/Mod/` directory.

Inside there are the instructions of what needs to be done. Basically, some lines need to be commented out in order to trigger the normalization of line endings. This was done by Bernd for some workbenches like Arch, Draft, Fem, Material, etc. He didn't touch other workbenches like Path and TechDraw because he expects the maintainers of these workbenches to take care of this when they see appropriate.

Something like this needs to be written
```
...
# Path/** -text
...
```

If there are special text-only files that must be converted, these need to be added to the list of extensions too.

Reply:
Thank you for the forum references and additional information.  It is much appreciated.